addons/website/static/src/snippets/s_masonry_block/s_masonry_block.xml

```diff
@@ -0,0 +1,252 @@
+<?xml version="1.0" encoding="utf-8"?>
+<template id="masonry_template">
+    <t t-name="website.s_template_masonry.default">
+        <div class="row">
+            <div class="col-lg-6 oe_img_bg text-center pb224 pt224" data-name="Block"
+                 style="background-image: url(/web/image/website.s_masonry_block_default_image_1);">
+                <p><br/></p>
+            </div>
+            <div class="col-lg-6 s_col_no_resize">
+                <div class="row h-100">
+                    <div class="col-lg-6 pt24 pb8 text-center bg-200" data-name="Block">
+                        <h3>A great title</h3>
+                        <p>And a great subtitle</p>
+                    </div>
+                    <div class="col-lg-6 pt24 pb8 text-center o_cc o_cc3" data-name="Block">
+                        <h3>A great title</h3>
+                        <p>And a great subtitle</p>
+                    </div>
+                    <div class="col-lg-6 pt24 pb8 text-center o_cc o_cc4" data-name="Block">
+                        <h3>A great title</h3>
+                        <p>And a great subtitle</p>
+                    </div>
+                    <div class="col-lg-6 pt24 pb8 text-center bg-200" data-name="Block">
+                        <h3>A great title</h3>
+                        <p>And a great subtitle</p>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </t>
```

Comment:
It would be the right way to do it for standard Odoo code but for website snippets I am not so sure. All our snippet are declared in qweb "server-side", I think those should too at least by consistence. That would also allow themes and community modules to extend those template as naturally as they would extend other snippets.

So move all of those next to the snippet definition in `addons/website/views/snippets/s_masonry_block.xml`. You then won't need the `onBuild` method I talked about: the snippet should directly be filled with the default template, using a t-call directive. Other advantage: the drag and drop preview you get when you you drag a snippet in the page without releasing your mouse will be accurate (not an empty block you move around until it is dropped in the page).

You can check `render_public_asset` to render "server-side" views from the JS.